interview-questions.md

```diff
@@ -8,36 +8,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 Researched answer:
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: GET(when using an show method), POST, PATCH/PUT and DELETE.
 
-Researched answer:
+Researched answer: POST requests to the create action: When creating a new resource, you must pass parameters to set the initial values for the resource's attributes. PATCH/PUT requests to the update action: When updating an existing resource, you must pass parameters to indicate which attributes to update and their new values. DELETE requests to the destroy action: When deleting a resource, you must pass a parameter to identify which resource to delete.
 
 3. Name three rails generator commands. What is created by each?
 
-Your answer:
+Your answer: model, controller and migration. The generate model creates a new table and when generating a model you need to build its structure when generating the model (column name and data type). Controller generates a controller file that is associated with a specific model to handle its CRUD actions. and Generate Migration is used for when you are going to make changes to your DB table that then need to be migrated for for the changes to take place. It also creates a paper trail of your changes.
 
 Researched answer:
+- generate model creates a new model file in the app/models directory, along with a migration file in the db/migrate directory that defines the necessary database changes to create the associated table. The model file contains Ruby code that defines the structure and behavior of the model, including any validations, associations, and other methods.
+
+- generate controllercreates a new controller file in the app/controllers directory, along with a set of view files in the app/views directory. The controller file contains Ruby code that defines the actions that the controller will handle, while the view files contain the HTML or other output that will be displayed to the user.
+
+- generate migration creates a new migration file in the db/migrate directory, which is used to make changes to the database schema. The migration file contains Ruby code that defines the necessary changes to the database, such as creating, modifying, or deleting tables or columns.
```

Comment:
Nice! I love the consideration of the files being created and def something to keep in mind.